src/cmdc/datasets/uscensus/census.py

```diff
@@ -0,0 +1,279 @@
+                msg += "geographies in the `VALID_GEOGRAPHIES` dict."
+                raise ValueError(msg)
+
+            out = "&for=" + VALID_GEOGRAPHIES[geography]
+
+        elif isinstance(geography, dict):
+            is_valid = "for" in geography.keys()
+            if not is_valid:
+                msg = "If you pass a dict, you must have keys 'for' and 'in'"
+                raise ValueError(msg)
+
+            out = "&for=" + geography["for"][0]
+            out += ":" + ",".join(map(str, geography["for"][1]))
+
+            if "in" in geography.keys():
+                out += "&in=" + geography["in"][0]
+                out += ":" + ",".join(map(str, geography["in"][1]))
+
+        elif isinstance(geography, tuple):
+            is_valid = geography[0] in VALID_GEOGRAPHIES.keys()
+
+            out = "&for=" + geography[0]
+            out += ":" + ",".join(map(str, geography[1]))
+
+        return out
+
+    def _get(self, columns, geography):
+        """
+        This is the hidden get method that actually fetches the data
+        from the US Census API.
+
+        Parameters
+        ----------
+        columns : list(string)
+            The data columns that you would like
+        geography : str, dict, or tuple
+            See documentation for `self.process_geography_args`
+        """
+        # Build up the request url
+        req_url = self.api_url + f"?key={self.key}"
+        req_url += self.process_column_args(columns)
+        req_url += self.process_geography_args(geography)
+
+        req = requests.get(req_url)
+
+        if req.status_code != 200:
+            print(req.status_code)
+            print(req.text)
+            msg = "API request failed... See printout above for more info"
+            raise ValueError(msg)
+
+        return req
+
+    def _process_get_json(self, columns, req_json):
+        """
+        Converts a particular request into a DataFrame
+
+        Parameters
+        ----------
+        columns : list
+            A list of the numeric columns (variables) of the table
+        req_json : dict
+            The json of the request from `self._get`
+
+        Returns
+        -------
+        df : pd.DataFrame
+            A DataFrame representation of the data in req_json
+        """
+        df = pd.DataFrame(data=req_json[1:], columns=req_json[0])
+        df[columns] = df[columns].apply(
+            lambda x: pd.to_numeric(x, errors="ignore")
+        )
+
+        return df
+
+    def get(self, columns, geography):
+        """
+        Retrieves the data from a particular US Census data source
+
+        Parameters
+        ----------
+        columns : list(string)
+            The data columns that you would like
+        geography : str, dict, or tuple
+            See documentation for `self.process_geography_args`
+        """
+        # Perform actual request... If it is a tract then we have to
+        # do the requests state by state...
+        if geography == "tract":
```

Comment:
Do we need to go down to the tract level? Meaning are any policy decisions being made at that level, or is county sufficient?

Reply:
There are some counties (admittedly, not many) that are reporting cases down to the tract level.

A county could potentially focus on particular tracts within their county if they notice that it looks particularly bad there. It didn't cost me much to incorporate the ability to ingest tract data so I evaluated benefit > cost.